### ISSUE TITLE
Emit peer locality label in metrics

### ISSUE THREAD

**Issue by dwj300** [istio/istio#53838]
(This is used to request new product features, please visit <https://github.com/istio/istio/discussions> for questions on using Istio)

**Describe the feature request**
We use locality-aware routing, but have failover enabled. We'd like to understand a) when and b) how much traffic fails over to cross AZ/locality. To do this, we'd want to exchange the peer-locality info via either metadata exchange, or the new metadata path, then add a label to the `istio_requests_total` metric (and probably others as well) for peer-locality.

**Describe alternatives you've considered**
We can use an augmented label (availability-zone + region) along with the `Reporter`, to identify the locality distribution of traffic between two services. If we see 1/3 outbound from three AZs (`reporter: source`) and 1/2 inbound from two AZs (reporter:destination`), we can derive that it is crossing the AZ boundary. But if the distribution is balanced, it could still be crossing the boundary, just evenly.

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by howardjohn** [comment]
Based on the context here I think you may already know this, but one note - you could, I believe, add a custom label with sourceLocality today (either with Istio natively, or OTEL/Prom/etc augmentation), so we really only need the remote.

https://github.com/tetratelabs/istio-cost-analyzer/blob/b9135a1c6a13c9808a7e195d8781bb1558d87b24/pkg/kube.go#L316 claims to do this, but I don't see how it can possibly work TBH

**Comment by howardjohn** [comment]
One way to do this for the outbound side would be to pull it from the EDS info. However, I don't see this exposed: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes. It also wouldn't solve the inbound side

**Comment by dwj300** [comment]
> Based on the context here I think you may already know this, but one note - you could, I believe, add a custom label with sourceLocality today (either with Istio natively, or OTEL/Prom/etc augmentation), so we really only need the remote.
> 
> https://github.com/tetratelabs/istio-cost-analyzer/blob/b9135a1c6a13c9808a7e195d8781bb1558d87b24/pkg/kube.go#L316 claims to do this, but I don't see how it can possibly work TBH

Yeah exactly - we just need the peer/remote. I looked at adding it to the HTTP metadata exchange (which would get us partially there), but it's tricky since it has to be able backwards compatible with old proxies. EDS would get us part of the way there. Isn't there another out of band metadata exchange channel?

**Comment by keithmattix** [comment]
> Isn't there another out of band metadata exchange channel?

There is for ambient (WDS), but still not sure that gets you what you want.
